app/scenes/HomeScreen/index.js

```diff
@@ -1,45 +1,110 @@
-import React, { useEffect } from 'react';
-import { Button, Platform, SafeAreaView, Text, TextInput } from 'react-native';
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  FlatList,
+  SafeAreaView,
+  Dimensions,
+  TextInput,
+  View,
+  Animated
+} from 'react-native';
 import { connect } from 'react-redux';
 import { compose } from 'redux';
-import { PropTypes } from 'prop-types';
+import PropTypes from 'prop-types';
 import { createStructuredSelector } from 'reselect';
 import { injectIntl } from 'react-intl';
 import styled from 'styled-components/native';
-import LogoWithInstructions from '../../components/molecules/LogoWithInstructions';
+import isEmpty from 'lodash/isEmpty';
+import debounce from 'lodash/debounce';
+import Logo from '../../components/molecules/Logo';
 import { selectSongName, selectSongsData, selectSongsError } from './selectors';
 import { homeContainerCreators } from './reducer';
+import If from '../../components/molecules/If';
+import SoundCard from '../../components/organisms/SoundCard';
+const { width } = Dimensions.get('window');
 
-const StyledBox = styled(Text)`
-  margin: 30px auto;
+const SearchBox = styled(TextInput)`
+  margin: 14px 16px 10px;
+  border-width: 0.5px;
+  border-color: #c4c3c3;
+  padding: 16px;
 `;
+
 function HomeScreen({
   songName,
   songsData,
   dispatchSongs,
-  dispatchClearSongsPlaylist
+  dispatchClearSongsPlaylist,
+  intl
 }) {
-  const instructions = Platform.select({
-    ios: 'Press Cmd+R to reload',
-    android: 'Double tap R on your keyboard to reload'
-  });
-  const buttonHandler = () => {
-    dispatchSongs('eminem');
+  const scrollX = useRef(new Animated.Value(0)).current;
+  const [index, setIndex] = useState(0);
+  const slider = useRef(null);
+
+  useEffect(() => {
+    if (songName && !songsData?.items?.length) {
+      dispatchSongs(songName);
+    }
+    // to set the index of current card
+    scrollX.addListener(({ value }) => {
+      const rawIndex = Math.round(value / width);
+      setIndex(rawIndex);
+    });
+  }, []);
```

Comment:
```suggestion
  return () => {
    scrollX.removeListener(listenerId)
  }
  }, []);
```

We could unregister the listener on unmount.

Checkout [removeListener](https://reactnative.dev/docs/animatedvaluexy#removelistener)